# IS362 Week 5 Assignment


Your task in this week’s assignment is to answer three questions:
1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12th, 2013, which New York area airport had the windiest weather?

I'll be answering the first 2 questions based on the data that was given:

https://github.com/hadley/nycflights13/tree/master/data-raw

I downloaded airports.csv first.

import pandas as pd

read_file = pd.read_csv('airports.csv')
read_file

## Northernmost Airport in the U.S.

In [5]:
#sort out the top 5 northenmost airports

read_file.sort_values(by ='lat', ascending = False).head(5)

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                 \N  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

Based on the output from sorting out the data by latitude, it shows that Dillant Hopkins Airport is the northernmost airport in the U.S.  I did a google search to confirm if this is correct.  I googled the latitude of Dillant Hopkins Airport in which is showed the following:

![title](pictures/dillant.png)

It seems like the data on the .csv file was incorrect.  They had the latitude and longitude of Dillant Hopkins Airport in the wrong columns.  Given that it is not Dillant, I googled the next airport on the list which is Wiley Post Rogers Memorial Airport.

![title](pictures/wiley.png)

So the answer is Wiley Post Rogers Memorial Airport as the northernmost aiport in the U.S.

## Easternmost Airport in the U.S.

In [6]:
#sort out the easternmost airport

read_file.sort_values(by ='lon', ascending = False).head(5)

faa                           name        lat         lon   alt  tz dst  \
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   

                  tzone  
1290  America/Anchorage  
942      Asia/Chongqing  
396      Asia/Chongqing  
417                  \N  
444    America/New_York

According to the sorted data, Eareckson Air Station is the easternmost airport in the U.S.  I confirmed if the information is correct on the dataset by doing a Google search.

![title](pictures/east.png)

This map shows Eareckson Air Station's location.  Although it looks it's in the west, it's actually in the east.

![title](pictures/map.png)

Eareckson Air Station is the easternmost airport in the U.S.

## Windiest Weather


To find out which airport had the windiest weather on February 12, 2013, I will be using the weather.csv file.


In [8]:
weather_file = pd.read_csv('weather.csv')
weather_file

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0        EWR  2013      1    1     0  37.04  21.92  53.97     230.0   
1        EWR  2013      1    1     1  37.04  21.92  53.97     230.0   
2        EWR  2013      1    1     2  37.94  21.92  52.09     230.0   
3        EWR  2013      1    1     3  37.94  23.00  54.51     230.0   
4        EWR  2013      1    1     4  37.94  24.08  57.04     240.0   
5        EWR  2013      1    1     6  39.02  26.06  59.37     270.0   
6        EWR  2013      1    1     7  39.02  26.96  61.63     250.0   
7        EWR  2013      1    1     8  39.02  28.04  64.43     240.0   
8        EWR  2013      1    1     9  39.92  28.04  62.21     250.0   
9        EWR  2013      1    1    10  39.02  28.04  64.43     260.0   
10       EWR  2013      1    1    11  37.94  28.04  67.21     240.0   
11       EWR  2013      1    1    12  39.02  28.04  64.43     240.0   
12       EWR  2013      1    1    13  39.92  28.04  62.21     250.0   
13       EWR  2013      1    1    14  39.92  28.04  62.21     260.0   
14       EWR  2013      1    1    15  41.00  28.04  59.65     260.0   
15       EWR  2013      1    1    16  41.00  26.96  57.06     260.0   
16       EWR  2013      1    1    17  39.20  28.40  64.93     270.0   
17       EWR  2013      1    1    18  39.20  28.40  64.93     330.0   
18       EWR  2013      1    1    19  39.02  24.08  54.68     280.0   
19       EWR  2013      1    1    20  37.94  24.08  57.04     290.0   
20       EWR  2013      1    1    21  37.04  19.94  49.62     300.0   
21       EWR  2013      1    1    22  35.96  19.04  49.83     330.0   
22       EWR  2013      1    1    23  33.98  15.08  45.43     310.0   
23       EWR  2013      1    2     0  33.08  12.92  42.84     320.0   
24       EWR  2013      1    2     1  32.00  15.08  49.19     310.0   
25       EWR  2013      1    2     2  30.02  12.92  48.48     320.0   
26       EWR  2013      1    2     3  28.94  12.02  48.69     320.0   
27       EWR  2013      1    2     4  28.04  10.94  48.15     310.0   
28       EWR  2013      1    2     5  26.96  10.94  50.34     310.0   
29       EWR  2013      1    2     6  26.06  10.94  52.25     330.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
26100    LGA  2013     12   29    18  42.80  37.40  81.07      70.0   
26101    LGA  2013     12   29    19  39.20  37.40  93.19      40.0   
26102    LGA  2013     12   29    20  41.00  39.02  92.59      40.0   
26103    LGA  2013     12   29    21  41.00  37.94  88.76     350.0   
26104    LGA  2013     12   29    22  41.00  39.20  93.24      50.0   
26105    LGA  2013     12   29    23  42.08  39.02  88.81     330.0   
26106    LGA  2013     12   30     0  42.80  37.40  81.07     320.0   
26107    LGA  2013     12   30     1  41.00  37.40  86.89     310.0   
26108    LGA  2013     12   30     2  42.80  37.40  81.07     320.0   
26109    LGA  2013     12   30     3  42.98  37.04  79.38     300.0   
26110    LGA  2013     12   30     4  42.98  35.06  73.39     310.0   
26111    LGA  2013     12   30     5  42.08  33.98  72.78     320.0   
26112    LGA  2013     12   30     6  42.08  33.98  72.78     250.0   
26113    LGA  2013     12   30     7  41.00  33.98  75.88     240.0   
26114    LGA  2013     12   30     8  42.98  33.98  70.30     270.0   
26115    LGA  2013     12   30     9  41.00  33.08  73.19       0.0   
26116    LGA  2013     12   30    10  42.98  33.08  67.81     250.0   
26117    LGA  2013     12   30    11  42.98  33.98  70.30     230.0   
26118    LGA  2013     12   30    12  44.06  35.06  70.42     240.0   
26119    LGA  2013     12   30    13  44.06  33.98  67.45     260.0   
26120    LGA  2013     12   30    14  44.06  33.08  65.07     260.0   
26121    LGA  2013     12   30    15  42.80  33.80  70.28     330.0   
26122    LGA  2013     12   30    16  41.00  28.40  60.54     340.0   
26123    LGA  2013     12   30    17  37.94  23.00  54.51     330.0   
26124    LGA  2013     12   30

In [13]:
#sort out data that has the date February 12, 2013

weather_sort = weather_file[(weather_file['year'] == 2013) & (weather_file['month'] == 2) & (weather_file['day'] == 12)]
weather_sort.sort_values(by='wind_speed', ascending = False).head(5)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1015     EWR  2013      2   12     8  39.02  26.96  61.63     260.0   
18433    LGA  2013      2   12     7  42.98  26.06  50.94     290.0   
1024     EWR  2013      2   12    17  44.06  26.06  48.87     270.0   
18444    LGA  2013      2   12    18  44.06  23.00  43.02     300.0   
9728     JFK  2013      2   12    13  42.08  24.08  48.56     280.0   

       wind_speed    wind_gust  precip  pressure  visib             time_hour  
1015   1048.36058  1206.432388     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18433    23.01560    26.485892     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1024     21.86482    25.161598     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18444    21.86482    25.161598     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
9728     20.71404    23.837303     0.0    1011.6   10.0  2013-02-12T13:00:00Z

According to the sorted data, EWR airport was the windiest airport in February 12, 2013 with a wind speed of 1048.36058.  However, this number is an outlier.  Without the outlier, the windiest airport on that date is LGA.